server/src/test/java/org/elasticsearch/index/mapper/GeoPointFieldMapperTests.java

```diff
@@ -192,6 +193,14 @@ public void testIgnoreZValue() throws IOException {
         assertThat(ignoreZValue, equalTo(false));
     }
 
+    public void testIndexParameter() throws Exception {
+        DocumentMapper mapper = createDocumentMapper(fieldMapping(b -> b.field("type", "geo_point").field("index", false)));
+        Mapper fieldMapper = mapper.mappers().getMapper("field");
+        assertThat(fieldMapper, instanceOf(GeoPointFieldMapper.class));
+        boolean searchable = ((GeoPointFieldMapper)fieldMapper).fieldType().isSearchable();
```

Comment:
We could also check that parsing omits the right fields, as in `KeywordFieldMapperTests#testDisableIndex()`.